Lib/socketserver.py

```diff
@@ -651,24 +652,32 @@ def process_request_thread(self, request, client_address):
         except Exception:
             self.handle_error(request, client_address)
         finally:
-            self.shutdown_request(request)
+            try:
+                self.shutdown_request(request)
+            finally:
+                thread = threading.current_thread()
+                with self._threads_lock:
+                    if self._threads and not thread.daemon:
+                        self._threads.remove(thread)
```

Comment:
I would prefer to ignore ValueError in case thread is already removed from _threads. It shouldn't happen, but I prefer to be extra careful.

Reply:
`_threads.remove()` now suppresses `ValueError`.